packages/react/src/components/fileInput/FileInput.tsx

```diff
@@ -0,0 +1,554 @@
+    hasFilesSelected && fileListId,
+  ]
+    .filter((text) => !!text)
+    .join(' ');
+
+  return (
+    <>
+      <InputWrapper {...wrapperProps}>
+        <div className={styles.fileInputContainer}>
+          {hasDragAndDrop && (
+            <>
+              <div
+                aria-hidden
+                onDragEnter={onDragEnter}
+                onDragOver={onDragEnter}
+                onDragLeave={onDragLeave}
+                onDrop={(event: React.DragEvent<HTMLDivElement>) => {
+                  const { dataTransfer }: { dataTransfer: DataTransfer } = event;
+                  onDragLeave(event);
+                  onFilesChange(Array.from(dataTransfer.files));
+                }}
+                className={classNames(styles.dragAndDrop, isDragOverDrop && styles.dragAndDropActive)}
+                ref={dropAreaRef}
+              >
+                <div className={styles.dragAndDropLabel}>
+                  <IconUpload aria-hidden />
+                  <span className={styles.dragAndDropLabelText}>{dragAndDrop.label}</span>
+                </div>
+              </div>
+              <div className={styles.dragAndDropHelperText}>{dragAndDrop.helperText}</div>
+            </>
+          )}
+          <div className={styles.fileInputWrapper}>
+            <Button
+              aria-hidden
+              variant="secondary"
+              iconLeft={<IconPlus aria-hidden />}
+              onClick={() => passButtonClickToInput()}
+              disabled={disabled}
+            >
+              {buttonLabel}
+            </Button>
+            <input
+              type="file"
+              ref={inputRef}
+              id={id}
+              disabled={disabled}
+              required={required}
+              aria-describedby={ariaDescribedBy}
+              className={styles.fileInput}
+              onChange={(event: ChangeEvent<HTMLInputElement>) => {
+                onFilesChange(Array.from(event.target.files));
+              }}
+              {...(accept ? { accept } : {})}
+              {...(multiple ? { multiple } : {})}
+            />
+          </div>
+        </div>
+      </InputWrapper>
+      <ul
+        id={fileListId}
+        ref={fileListRef}
+        tabIndex={-1}
+        className={styles.fileList}
+        aria-label={
+          hasFilesSelected ? getFileListAriaLabel(language, selectedFiles.length) : getNoFilesAddedMessage(language)
+        }
+      >
+        {selectedFiles.map((file: File, index: number) => (
+          <li
+            key={file.name}
+            className={styles.fileListItem}
+            tabIndex={-1}
+            ref={(el) => {
+              if (el && fileListRef.current && fileListFocusIndexRef.current === index) {
+                el.focus();
+              }
+            }}
+          >
+            {file.type.startsWith('image') ? <IconPhoto aria-hidden /> : <IconDocument aria-hidden />}
```

Comment:
Not sure if it's easy/worth to do, but it might look better in some cases if the icon, file size and remove button would be vertically centered. If the filename is long it can span to multiple lines like this:

![image](https://user-images.githubusercontent.com/2777633/131997229-31d0bb7b-ae52-4a4d-b82f-0c8a2dc17d4b.png)

Reply:
They should be "vertically centered", but I have to check this 🤔

Reply:
- [ ] check the alignment of icon, size text and remove button

Reply:
It might be that I misunderstood but they are already aligned 🤔

Reply:
Yeah I was thinking something like this:
![131997229-31d0bb7b-ae52-4a4d-b82f-0c8a2dc17d4b](https://user-images.githubusercontent.com/2777633/132496242-0d620405-82bd-47ed-9b5d-103598a07203.png)

->

![Vertically aligned](https://user-images.githubusercontent.com/2777633/132496298-07d2ecfb-3cab-4af7-9621-740382c2e023.png)

Reply:
Though it might even look better without the vertical alignment. Specially if the filename spans to more lines than 2, it can look weird if the icon is centered

Reply:
Yes, it was originally implementation but it got fixed when the names spanned to multiple lines

Reply:
- [ ] fix alignment